### DISCUSSION TITLE
Support for refresh of Non File Nodes

### DISCUSSION THREAD

**Discussion by vibsid0986** [eclipse-theia/theia#13965] [Q&A]
Hi Team,

Taking reference from this [thread](https://github.com/eclipse-theia/theia/discussions/13266), I was able to successfully implement custom property view of CRUD operations on Non-file nodes, I implemented a new file system provider (scheme -> nonFile:// ) for non file nodes. Everything is working as expected, however navigator does not refresh automatically whenever I delete or create any non file node. Request you to help me with understanding how to do it. Your help would be deeply appreciated.

Thanks in advance.

**Comment by msujew** [comment]
Hey @vibsid0986,

have you implemented the `onDidChangeFile` event correctly in your file system provider? See also [this implementation for the collaboration mode](https://github.com/eclipse-theia/theia/blob/38d180f131f13d7b5337c715e7ef0ccac176a993/packages/collaboration/src/browser/collaboration-file-system-provider.ts#L70-L72) that allows to trigger the event programmatically.
